packages/amplify-category-function/src/provider-utils/awscloudformation/index.ts

```diff
@@ -139,6 +141,67 @@ async function addLayerResource(
 }
 
 export async function updateResource(context, category, service, parameters, resourceToUpdate) {
+  // load the service config for this service
+  const serviceConfig: ServiceConfig<FunctionParameters> | ServiceConfig<LayerParameters> = supportedServices[service];
+  const BAD_SERVICE_ERR = `amplify-category-function is not configured to provide service type ${service}`;
+  if (!serviceConfig) {
+    throw BAD_SERVICE_ERR;
+  }
+  switch (service) {
+    case ServiceName.LambdaFunction:
+      return updateFunctionResource(context, category, service, parameters, resourceToUpdate);
+    case ServiceName.LambdaLayer:
+      return updateLayerResource(context, category, service, serviceConfig, parameters);
+    default:
+      throw BAD_SERVICE_ERR;
+  }
+}
```

Comment:
Instead of having this switch statement here and duplicated in the addResource method above, we should consider having a different controller here: https://github.com/aws-amplify/amplify-cli/blob/lambdaLayers/packages/amplify-category-function/src/provider-utils/supported-services.ts#L20,L29 so that once we have determined the service config no more disambiguation is necessary

Reply:
Do you mean I should changed it as LambdaController and Layer controller for the add/update function respectively?

Reply:
Yeah, like instead of using the same controller and having switch / if statement within each function to switch behavior, we should just have two separate controllers referenced in the service config